src/media/dto/file.dto.ts

```diff
@@ -1,4 +1,16 @@
 import { Readable } from 'stream';
+import { ApiProperty } from '@nestjs/swagger';
+
+/**
+ * This DTO is only used for properly typing our API. We cannot use the actual
+ * UploadFileDto there, as this already unwraps our uploaded file.
+ *
+ * See https://docs.nestjs.com/openapi/operations#file-upload
+ */
+export class SingleFileUploadDto {
```

Comment:
Isn't there another way for this? Like adding the class inline in the swagger annotation? Like this maybe?

@ApiBody({
    description: 'File to be uploaded',
    type: {
       @ApiProperty({ type: 'string', format: 'binary' })
        file: any;
    }
  })

Reply:
Yes, you could specify the type inline, but then you would not get a proper schema definition in Swagger UI. By adding the DTO, we actually get proper schema definitions for all of our API.

However, I changed the comment slightly because it did not make sense - it's perfectly fine doing it this way, because this is the actual DTO our app gets, while the `@UploadedFile` decorator performs the magic and changes its type. See https://github.com/gipfeli-io/gipfeli-api/pull/103